thorlcr/master/thgraphmanager.cpp

```diff
@@ -1456,63 +1457,153 @@ void thorMain(ILogMsgHandler *logHandler, const char *wuid, const char *graphNam
+                                WARNLOG("Discarding agentless job: wuid=%s, graph=%s", currentWuid.str(), currentGraphName.str());
+                                currentWuid.clear();
                             }
                             else
                             {
-                                switch (w->getState())
+                                SCMStringBuffer jobVersion;
+                                workunit->getDebugValue("platformVersion", jobVersion);
+                                if (jobVersion.length() && !thorQueue)
                                 {
-                                    case WUStateRunning:
-                                        w->setState(WUStateWait);
-                                        break;
-                                    case WUStateAborting:
-                                    case WUStateAborted:
-                                    case WUStateFailed:
-                                        break;
-                                    default:
-                                        w->setState(WUStateFailed);
-                                        break;
+                                    Owned<IException> exception = MakeStringException(TE_AbortException, "Custom platformVersion with multiJobLinger=false not supported");
+                                    relayWuidException(workunit, exception);
+                                }
+                                else
+                                {
+                                    bool mismatch = false;
+                                    const char *imagePattern = getenv("imagePattern");
+                                    const char *imageVersion = nullptr;
+                                    if (isEmptyString(imagePattern))
+                                        IWARNLOG("imagePattern not set in environment");
+                                    else
+                                    {
+                                        imageVersion = strrchr(imagePattern, ':');
+                                        if (!imageVersion)
+                                            IWARNLOG("imagePattern has unexpected format: %s", imagePattern);
+                                        else
+                                        {
+                                            imageVersion++;
+                                            if (jobVersion.length())
+                                            {
+                                                if (0 == runtimePlatformVersion.length()) // 1st time
+                                                {
+                                                    // check if jobVersion is same as original helm chart image
+                                                    // if it is, then we are not a custom runtime version
+                                                    if (!streq(jobVersion.str(), imageVersion))
```

Comment:
This needs to unconditionally set runtimePlatformVersion.  Otherwise setting it to the current version and then a different version will cause problems.

Reply:
you're right, well spotted.